waku/v2/node/rest/store/handlers.nim

```diff
@@ -0,0 +1,251 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  stew/results,
+  chronicles,
+  json_serialization,
+  presto/route
+import
+  ../../../../common/base64,
+  ../../../protocol/waku_message/topics/content_topic,
+  ../../../protocol/waku_store/common,
+  ../../../utils/time,
+  ../../waku_node,
+  ../../peer_manager,
+  ../responses,
+  ../serdes,
+  ./types
+
+export types
+
+logScope:
+  topics = "waku node rest store_api"
+
+const futTimeout* = 5.seconds # Max time to wait for futures
+
+# Queries the store-node with the query parameters and
+# returns a RestApiResponse that is sent back to the api client.
+proc makeHistoryQuery(selfNode: WakuNode,
+                      histQuery: HistoryQuery,
+                      storePeer: RemotePeerInfo):
+
+                      Future[RestApiResponse] {.async.} =
+
+  let queryFut = selfNode.query(histQuery, storePeer)
+  if not await queryFut.withTimeout(futTimeout):
+    const msg = "No history response received (timeout)"
+    error msg
+    return RestApiResponse.internalServerError(msg)
+
+  let res = queryFut.read()
+  if res.isErr():
+    const msg = "Error occurred in queryFut.read()"
+    error msg, error=res.error
+    return RestApiResponse.internalServerError(msg)
+
+  let storeResp = res.value.toStoreResponseRest()
+  let resp = RestApiResponse.jsonResponse(storeResp, status=Http200)
+  if resp.isErr():
+    const msg = "Error building the json respose"
+    error msg, error=resp.error
+    return RestApiResponse.internalServerError(msg)
+
+  return resp.get()
+
+# Converts timestamp in string format to Timestamp object
+proc toTimestamp(input: string): Timestamp =
+  try:
+    return Timestamp(parseInt(input))
+  except CatchableError as e:
+    error "Error converting to Timestamp ", error=e.msg
+    return Timestamp(0)
+
+# Checks whether the string represents a valid positive Timestamp
+proc isValidTimestamp(self: Option[string]): bool =
+  if self.isSome() and self.get() != "":
+    try:
+      let time = parseInt(self.get())
+      return time > 0
+    except CatchableError as e:
+      debug "Problem in isValidTimestamp ", error=e.msg
+
+  return false
+
+# Checks whether the peerAddr parameter represents a valid p2p multiaddress.
+# The param must be in the format `(ip4|ip6)/tcp/p2p/$peerId` but URL-encoded
+proc parsePeerAddr(peerAddr: string): Result[RemotePeerInfo, string] =
```

Comment:
Yes, will be in _utils/peers.nim_ in the next commit. We are planning to modify the current `parseRemotePeerInfo` in further PR's so that it doesn't raise any exception.